File: 118-pascals-triangle/118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> res;
        int n=0;
        res.push_back({1});
        while(--numRows){
            vector<int>curr;
            curr.push_back(1);
            
            int i=1;
            while(i<res[n].size()){
                curr.push_back(res[n][i-1] + res[n][i]);
                i++;
            }
            curr.push_back(1);
            res.push_back(curr);
            n++;
        }
        
        return res;
    }
};

File: 118-pascals-triangle/118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int nums) {
        List<List<Integer>> res=new ArrayList<>();
        
        List<Integer> first=new ArrayList<Integer>();
        first.add(1);
        res.add(first);
        
        for(int i=2;i<=nums;i++){
            List<Integer> curr=res.get(res.size()-1);
            List<Integer> n=new ArrayList<Integer>();
            n.add(1);
            for(int j=1;j<curr.size();j++){
                n.add(curr.get(j)+curr.get(j-1));
            }
            n.add(1);
            res.add(n);
        }
        
        return res;
    }
}